dashboard/views.py

```diff
@@ -1698,9 +1698,68 @@ def marshal(request):
     return render(request, 'marshal.html', context)
 
 
+def marshal_mab_edit_candidate(request):
+    initial = {'candidate': request.session.pop('candidate', None)}
+    form = MABEditCandidateForm(request.POST or None, initial=initial)
+
+    if request.method == 'POST':
+        if form.is_valid():
+            request.session['candidate'] = form.cleaned_data['candidate'].pk
+            return HttpResponseRedirect(reverse('dashboard:marshal_mab_edit'))
+
+    context = {
+        'form': form,
+    }
+
+    return render(request, 'marshal-mab-edit-candidate.html', context)
+
+
+def marshal_mab_edit(request):
+    candidate = Brother.objects.get(pk=request.session.get('candidate', None))
+    check_all = request.session.pop('check_all', False)
+
+    mab_form_list = []
+
+    brothers = Brother.objects.filter(brother_status='1')
+
+    for counter, brother in enumerate(brothers):
+        form = MeetABrotherEditForm(request.POST or None, prefix=counter+1, brother=brother.pk, mab_exists=MeetABrother.objects.filter(brother=brother, candidate=candidate).exists())
+        if check_all:
+            form.fields['update'].initial = True
+        mab_form_list.append(form)
+
+    if request.method == 'POST':
+        if 'submit' in request.POST:
+            if forms_is_valid(mab_form_list):
+                for counter, form in enumerate(mab_form_list):
+                    instance = form.cleaned_data
+                    if instance['update']:
+                        mab, created = MeetABrother.objects.get_or_create(candidate=candidate, brother=brothers[counter], completed=True, week=datetime.datetime(1000, 1, 1))
+                        if created:
+                            mab.save()
+                    elif not instance['update']:
+                        try:
+                            MeetABrother.objects.get(candidate=candidate, brother=brothers[counter], completed=True, week=datetime.datetime(1000, 1, 1)).delete()
+                        except MeetABrother.DoesNotExist:
+                            break
+                return HttpResponseRedirect(reverse('dashboard:marshal_mab_edit_candidate'))
+        if 'check_all' in request.POST:
+            request.session['check_all'] = True
+            return HttpResponseRedirect(reverse('dashboard:marshal_mab_edit'))
+        if 'go_back' in request.POST:
+            return HttpResponseRedirect(reverse('dashboard:marshal_mab_edit_candidate'))
+
+    context = {
+        'candidate': candidate,
+        'mab_form_list': mab_form_list,
+    }
+
+    return render(request, 'marshal-mab-edit.html', context)
+
+
 def meet_a_brother(request):
     candidates = Brother.objects.filter(brother_status=0)
-    weeks = MeetABrother.objects.all().order_by('-week').values_list('week', flat=True).distinct
+    weeks = MeetABrother.objects.filter(week__range=(datetime.date(2000, 1, 1), datetime.datetime.now())).order_by('-week').values_list('week', flat=True).distinct
```

Comment:
again, these magic numbers should be defined somewhere.  one option if the 1's go with the 2000 is to define

```
THOUSAND_CONSTANT_NAME = 1000, 1, 1
TWO_THOUSAND_CONSTANT_NAME = 2000, 1, 1
```

and then when you pass these in, just do

```
datetime.datetime(*THOUSAND_CONSTANT_NAME)
datetime.date(*TWO_THOUSAND_CONSTANT_NAME)
```

also why are you switching between using date and datetime (this might be correct, i just noticed the change and thought it was weird, i dont remember the difference between the functions)